samples/certificate_provisioning/boards/nrf52840dk_nrf52840.conf

```diff
@@ -0,0 +1,18 @@
+# WiFi
+CONFIG_WIFI=y
+CONFIG_WIFI_ESP_AT=y
+CONFIG_WIFI_ESP_AT_MDM_RX_BUF_COUNT=40
+
+CONFIG_NET_L2_WIFI_SHELL=y
+CONFIG_GOLIOTH_SAMPLE_WIFI=y
+
+# General config
+CONFIG_HEAP_MEM_POOL_SIZE=4096
+
+# MCU Manager
+CONFIG_MCUBOOT_BOOTUTIL_LIB=y
+CONFIG_MCUMGR_MGMT_NOTIFICATION_HOOKS=y
+CONFIG_MCUMGR_GRP_FS_FILE_ACCESS_HOOK=y
```

Comment:
These symbols were added to remove this compiler warning that appears when the device is flashed (I previously tried to replicate it during the build step but it's only present during the flashing step for some baffling reason):

```
CMake Warning at /home/mike/golioth-zephyr-workspace/zephyr/subsys/mgmt/mcumgr/grp/fs_mgmt/CMakeLists.txt:26 (message):
  Note: MCUmgr file system management is enabled but file access hooks are
  disabled, this is an insecure configuration and not recommended for
  production use, as all files on the filesystem can be manipulated by a
  remote device.  See
  https://docs.zephyrproject.org/latest/services/device_mgmt/mcumgr_callbacks.html
  for details on enabling and using MCUmgr hooks.
```

I haven't seen this message when using the nRF9160dk so I thought it best to put it in the board conf.

Would it help to use a different comment? Like:

```
# Use secure configuration file access options for nRF52840
```

Reply:
I'm not sure why the warning is only showing up when flashing the nrf52840dk, but it applies to all boards. For a sample, there's not really any problem with this warning, other than being annoying to users. In fact, even if it weren't a sample there wouldn't be an issue, because the only files in the filesystem are ones that we want to expose over `mcumgr`. Blocking access to other files doesn't actually make the device more secure (for this application).

We can fix it or we can leave it. I'm of the opinion that the warning is fine, and we should just ignore it, but it's not a strong opinion. If we decide to fix it, we should not only enable `CONFIG_MCUMGR_MGMT_NOTIFICATION_HOOKS` and `CONFIG_MCUMGR_GRP_FS_FILE_ACCESS_HOOK` in `prj.conf`, but actually implement the fs access hook to only allow access to the files in question, for completeness. Enabling the hook but not implementing is equivalent to not enabling it, save for silencing the warning, and doing so could set a bad example for users. The reason I think we should leave the warning is because the point of the sample is _not_ to show how to use `mcumgr` in general, and adding this extra code could unnecessarily complicate the sample. And if we decide to fix it, it should probably be in a separate ticket/PR, since this affects all boards.

Separate question: What do we need `CONFIG_MCUBOOT_BOOTUTIL_LIB` for?

Reply:
Okay, I agree we should leave these symbols out and be okay with the warning.

> Separate question: What do we need CONFIG_MCUBOOT_BOOTUTIL_LIB for?

I think I made a mistake when looking up dependencies. I think I was searching for hooks and looked at `BOOT_IMAGE_ACCESS_HOOKS` by mistake which depends on `MCUBOOT_BOOTUIL_LIB`. Sorry about that.